src/js/popup.js

```diff
@@ -339,6 +342,99 @@ $(document).ready(function () {
         });
     });
 
+    $('#customDomains').click(function () {
+        dmUIClick('customDomains');
+        if (!$(this).hasClass('disabled')) {
+            $('.custom-domain-list').toggle();
+            $('main').toggle();
+        } else {
+            Materialize.toast(disabledOptionToast, 2000, 'toastNotification');
+        }
+    });
+
+    $('#customDomainsBackButton').click(function () {
+        dmUIClick('customDomainsBackButton');
+        $('.custom-domain-list').toggle();
+        $('main').toggle();
+    });
+
+    $('#clearCustomDomains').click(function () {
+        dmUIClick('clearCustomDomains');
+        // remove all added domains:
+        chrome.runtime.sendMessage({
+            JDTIfunction: 'removeDomain',
+            domainName: '',
+            removeAll: true
+        }, function (response) {
+            if (response.status === 'success') {
+                loadTemplateEditor();
+                Materialize.toast('Domains successfully removed', 2000, 'toastNotification');
+            } else {
+                if (response.message) {
+                    dmError('clearCustomDomains', response.message);
+                    Materialize.toast(response.message, 2000, 'toastNotification');
+                } else {
+                    dmError('clearCustomDomains', 'generic');
+                    Materialize.toast('Something went wrong. Please try again.', 2000, 'toastNotification');
+                }
+            }
+        });
+    });
+
+    $('#customDomainInput').keyup(function (event) {
+        if (event.keyCode === 13) {
+            $('#customDomainInputButton').click();
+        }
+    });
+
+    $('#customDomainInputButton').click(function () {
+        dmUIClick('customDomainInputButton');
+        var domainName = $('#customDomainInput').val();
+
+        chrome.runtime.sendMessage({
+            JDTIfunction: 'addDomain',
+            domainName: domainName
+        }, function (response) {
+            if (response.status === 'success') {
+                $('#customDomainInput').val('');
+                loadTemplateEditor();
+                Materialize.toast('Domain successfully added', 2000, 'toastNotification');
+            } else {
+                if (response.message) {
+                    dmError('customDomainInputButton', response.message);
+                    Materialize.toast(response.message, 2000, 'toastNotification');
+                } else {
+                    dmError('customDomainInputButton', 'generic');
+                    Materialize.toast('Something went wrong. Please try again.', 2000, 'toastNotification');
+                }
+            }
+        });
+    });
+
+    $(document).on('click', '.customDomainRemoveButton', function () {
```

Comment:
Let's stick with the `$('.customDomainRemoveButton').click(function () {` notation. Also `-` (dash) case for classes.

Reply:
:+1: for the - (dash) case for classes.

it's `(document).on` for a reason. I've spent a nontrivial amount of time trying to make it work with `.click()`. However, I don't think it's possible. If you check the `removeSingleTemplate` section around line 665, you'll see that it also uses `(document.on)`, and there's a comment: 

``` // Because the template editing section is dynamically build, need to monitor document rather then the classes directly ```

The domain list is dynamically built similar to the template editing section.